lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1418,8 +1418,32 @@ SmallVector<Direction> PrimitiveOp::portDirections() {
 
 SmallVector<DictionaryAttr> PrimitiveOp::portAttributes() {
   SmallVector<DictionaryAttr> portAttributes;
-  for (size_t i = 0; i < getReferencedPrimitive().getAllPorts().size(); ++i)
-    portAttributes.push_back(DictionaryAttr());
+  OpBuilder builder(getContext());
+  auto cleanAttrs = [&](DictionaryAttr dict){
+    if (dict) {
+      llvm::SmallVector<NamedAttribute> attrs;
+      for (NamedAttribute attr : dict) {
+        Dialect *dialect = attr.getNameDialect();
+        if (dialect != nullptr && isa<CalyxDialect>(*dialect)) {
+          StringRef name = attr.getName().strref();
+          StringAttr newName =
+              builder.getStringAttr(std::get<1>(name.split(".")));
```

Comment:
Can we add a comment explaining where this logic is coming from? I imagine it's just removing the `calyx.` prefix?